### PR TITLE
Report view & Router tests + cleanup

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#10460](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10460)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

- Adds report view unit test
- Enhances Router tests
- Fixes tests warnings
- Re-enabled linting on ClientApp/test files

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/components/shared/hgButton.vue [hunk 1]
```diff
@@ -129,6 +129,27 @@ export default class HgButtonComponent extends Vue {
             text-decoration-color: $hg-button-secondary-hover-text;
             text-decoration-line: underline;
         }
+
+        &.selected {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/shared/hgButton.vue:133]: Added selected secondary button

File: Apps/WebClient/src/ClientApp/src/assets/scss/_variables.scss [hunk 2]
```diff
@@ -60,6 +60,17 @@ $hg-button-secondary-disabled: white;
 $hg-button-secondary-disabled-border: $hg-button-disabled;
 $hg-button-secondary-disabled-text: $hg-button-disabled;
 
+$hg-button-secondary-selected: #0092f1;
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/assets/scss/_variables.scss:63]: selected secondary button

**Comment by FuriousLlama** [comment]
Continuation of https://github.com/bcgov/healthgateway/pull/2255

File: Apps/WebClient/src/ClientApp/test/stubs/store/options.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import immunizationStub from "./immunizationStub";
+import laboratoryStub from "./laboratoryStub";
+import medicationStub from "./medicationStub";
+import navbarStub from "./navbarStub";
+import noteStub from "./noteStub";
+import timelineStub from "./timelineStub";
+import userStub from "./userStub";
+
+@injectable()
+export default class StoreOptionsStub implements GatewayStoreOptions {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/stubs/store/options.ts:23]: We have both `options.ts` and `storeOptionsStub.ts` and need to remove one.  I'd prefer keeping `storeOptionsStub.ts`.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/test/stubs/store/options.ts (hunk 3 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/stubs/store/options.ts:23]: agreed, is part of the automated merge

Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/test/stubs/store/options.ts:23]: I see both files still in the latest commit on this branch:

https://github.com/bcgov/healthgateway/tree/features/10460_b/Apps/WebClient/src/ClientApp/test/stubs/store

Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/stubs/store/options.ts]: Manually removed it now.

**Review by MikeLyttle — DISMISSED** [review]

**Comment by MikeLyttle** [comment]
When this is merged, *please* use the squashing option. 😄

File: Apps/WebClient/src/ClientApp/test/router.test.ts [hunk 4]
```diff
@@ -1,10 +1,12 @@
+// eslint-disable-next-line
+import container from "@/plugins/inversify.container";
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/router.test.ts:2]: Why does this needed to be imported first?  And why does reports.test.ts import `inversify.config` first instead?

Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/router.test.ts:2]: Inversify (DI library) needs to be imported pretty early before other definitions come into play to make sure the dependency injection works properly. This is also why reflect-metadata needs to be imported early on and only once

**Review by MikeLyttle — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**